styles/Canonical/014b-Numbers-with-four-or-more-digits-must-have-comma-separators.yml

```diff
@@ -0,0 +1,8 @@
+extends: existence
+message: Numbers greater than 999 should have a comma separator - '%s'
+link: https://docs.ubuntu.com/styleguide/en#numbers
+scope: raw
+ignorecase: true
+level: warning
+tokens:
+  - ([\d]{4,})
```

Comment:
@akcano @SecondSkoll This is a valid point, but are we okay with not flagging an actual 4-digit number in the wrong format (example: 2240), in favor of not wanting to generate warnings for years?

If not, is there a way to ignore specific instances either inline or via configuration?

Reply:
Can I ask if it is necessary to target `raw` scope? I'm just imagining applying this rule to HTML and getting thousands of errors. Wouldn't it be sufficient to target summary/headings/tables?

Reply:
The `summary` scope does not seem to work - when I tested locally and using the [online compiler](https://studio.vale.sh/) - it didn't detect any positives with the given test file. So I have defined select scopes that should cover heading, body (Sentences), tables.

Let me know what you think @evilnick @akcano @SecondSkoll

Reply:
> This is a valid point, but are we okay with not flagging an actual 4-digit number in the wrong format (example: 2240), in favor of not wanting to generate warnings for years?

I'm actually quite concerned about the years - the rule will fire on every single page (at least for RTD) because of the copyright year.
I would actually ask the other way around: Are we actually concerned about 4-digit numbers? Does it make a difference for readability if someone writes 1000 or 1,000?
My suggestion would be to only enforce the comma for 5- or more digit numbers.

Reply:
@ru-fu I see your point but want to understand more about the files on which the rules will be applied (Nick mentioned HTML too). Is it not limited to markdown or RST files? If so, wouldn't the redefined scope help?

Reply:
True - I think the plan is to mainly run on Markdown and RST.

Reply:
If we are applying only to source files (markdown and RST), the current scope should take care of that (except of course, the years that we actually use in text) and the level is kept at `warning` (we can demote it to `suggestion` even as Michael said but that's not going to prevent the flagging. 

However, I do not have a strong opinion on this as I was just trying to implement the scope that was defined. So I would like others' opinion on whether we should apply this for 5- or more digit numbers.

Reply:
Just as an FYI, we should also try and make the rules work on HTML.
 - marketing are probably going to end up using this ruleset too
 - I would like it to be something we can publish as a general set of rules for people to use, like the MS and Google ones
 
 If 5 or more numbers gets around a lot of pain that is probably okay. It doesn't change the style guide, just our nagging

Reply:
@ru-fu @evilnick I have updated the rule for 5- or more digit numbers. I guess we can see the rule in action and then refine as we need.